libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/core/http/HttpUtil.java

```diff
@@ -51,33 +52,26 @@ public HttpUtil(String ua) {
         userAgent = ua;
     }
 
-    private URLConnection openConnectionWithProxy(URL url) throws IOException {
-        Proxy proxy = Proxy.NO_PROXY;
-
-        ProxySelector proxySelector = ProxySelector.getDefault();
-        if (proxySelector != null) {
-            URI uri;
-            try {
-                uri = url.toURI();
-            } catch (URISyntaxException e) {
-                IOException ioe = new IOException(url.toString());
-                ioe.initCause(e);
-                throw ioe;
-            }
-
-            List<Proxy> proxies = proxySelector.select(uri);
-            if (proxies != null && !proxies.isEmpty()) {
-                proxy = proxies.get(0);
-            }
+    private URLConnection mozOpenConnection(URL l_url) throws IOException {
+        try {
+            // Try to use the highest possible TLS protocol that this platform supports.
+            SSLContext sslcontext = SSLContext.getInstance(GlobalConstants.DEFAULT_PROTOCOLS[0]);
+            sslcontext.init(null,
+                    null,
+                    null);
+            NoSSLv3SocketFactory factory = new NoSSLv3SocketFactory(sslcontext.getSocketFactory());
+            HttpsURLConnection.setDefaultSSLSocketFactory(factory);
+            return l_url.openConnection();
```

Comment:
HTTPConnection object go through 2 phases - openConnection() and connect().  Between the two you can set options like timeouts on the connection.